### PR TITLE
Standup rich text

### PR DESCRIPTION
**Pull request by GALTdea**
## What's the change?
- Adds Rich text ability to stand up form
## What key workflows are impacted?
It gives users filling out stand up updates "Rich Text" 
## Demo / Screenshot
Screenshot 2023-11-07 at 11 01 05 AM

Screenshot 2023-11-07 at 11 01 19 AM

## Checklist before requesting a review

Please delete items that are not relevant.

- [x] Did you consider risks around security, performance, etc.?
- [x] Related to readability and consistency: Did you add comments and/or documentation? (README, Notion, etc.)
- [x] Is everything elevated to the highest level? (e.g., can logic be moved out of view into controllers, or out of controllers into models?)
- [x] Is there any linting that needs to be executed?
- [x] Did you leave helpful inline PR comments for the reviewer(s)?

### CODE REVIEW

File: app/controllers/standup_meetings/completions_controller.rb
Comment (toyhammered) [inline app/controllers/standup_meetings/completions_controller.rb:21]: how come we have both of these now? Seems like too many params duplicated?

**Comment by JoshDevHub** [comment]
I think it'd be cool to see a screenshot of what it looks like when viewing someone's standup meeting content.

File: app/controllers/standup_meetings/completions_controller.rb
Comment (GALTdea) [reply app/controllers/standup_meetings/completions_controller.rb]: Good point, fixed!

**Comment by GALTdea** [comment]
> I think it'd be cool to see a screenshot of what it looks like when viewing someone's standup meeting content.

Yes, good point. Just added

File: lib/tasks/migrate_fields_to_rich_text.rake
Comment (toyhammered) [inline lib/tasks/migrate_fields_to_rich_text.rake:12]: i'm kind of confused as to what this exactly does?

Comment (JoshDevHub) [reply lib/tasks/migrate_fields_to_rich_text.rake:12]: This assigns the information that's currently in the `today_work_description`, `yesterday_work_description`, and `blockers_description` columns and assigns them as new `action_rich_texts` records associated with the standup meeting.

The new `has_rich_text` associations are set with the same name as the old columns. I also didn't expect this to work, but I just ran it locally and it does lol. I guess there's something funky with the way the reads/writes work regarding rich text associations having the same name as a column.

I probably would've done something like
- Introduce rich text associations with a different name than the current columns
- Move old column's information over to rich text
- Remove old columns (this should be done anyway, albeit in a separate PR from this one of course)
- If I wanted the rich text association names to be the same as the old column names, write a new task that does `ActionText::RichText.where(record_type: "StandupMeeting")` to pull the standup rich text records and then change their `name` attributes to match the old column names. And then make that adjustment in the standup meeting model file

That's all more complicated than what Gustavo has here, but also makes more sense to me lol.

Comment (toyhammered) [reply lib/tasks/migrate_fields_to_rich_text.rake:12]: > This assigns the information that's currently in the `today_work_description`, `yesterday_work_description`, and `blockers_description` columns and assigns them as new `action_rich_texts` records associated with the standup meeting.
> 
> The new `has_rich_text` associations are set with the same name as the old columns. I also didn't expect this to work, but I just ran it locally and it does lol. I guess there's something funky with the way the reads/writes work regarding rich text associations having the same name as a column.
> 
> I probably would've done something like
> 
> * Introduce rich text associations with a different name than the current columns
> * Move old columns' information over to rich text
> * Remove old columns (this should be done anyway, albeit in a separate PR from this one of course)
> * If I wanted the rich text association names to be the same as the old column names, write a new task that does `ActionText::RichText.where(record_type: "StandupMeeting")` to pull the standup rich text records and then change their `name` attributes to match the old column names. And then make that adjustment in the standup meeting model file to the `has_rich_text` associations.
> 
> That's all more complicated than what Gustavo has here, but also makes more sense to me lol.

What you just explained above is exactly how I would have done it also, but if this works, thats fine by me lol

**Review by toyhammered — APPROVED** [review]
I'll try to merge in and deploy to prod tomorrow + run the rake task.

File: lib/tasks/migrate_fields_to_rich_text.rake
Comment (GALTdea) [reply lib/tasks/migrate_fields_to_rich_text.rake:12]: Sorry @toyhammered, didn't get a chance to address your question. But Thanks @JoshDevHub for getting to it! 
Also, forgot to submit a chained pr with the migration to remove the old fields. I'm just going to wait until this one is merged!

**Pull request merged**
